src/drivers/usb/hc/ohci/ohci_pci.c

```diff
@@ -88,6 +88,10 @@ static void *ohci_ed_alloc(struct usb_endp *ep) {
 	struct ohci_td *sentinel_td = ohci_td_alloc();
 	struct ohci_ed *ed = pool_alloc(&ohci_eds);
 
+	if (!sentinel_td || !ed) {
+		return NULL;
```

Comment:
You might have to free `sentinel_td`

Reply:
That's right. Fixed.

Reply:
I'm agree with @Machiry you have to free allocated object if one of pointers are not null.
I think it's better to check them separately


```	struct ohci_td *sentinel_td;
	struct ohci_ed *ed; = pool_alloc(&ohci_eds);
	
 	sentinel_td = ohci_td_alloc();
	if (!sentinel_td ) {
	return NULL;
	}
        ed = pool_alloc(&ohci_eds);
        if (!ed) {
       ohci_td_free(sentinel_td);
       }
       ```